coffee_maker_navigator_2/lib/di/di/dependency_container_manager.dart

```diff
@@ -0,0 +1,137 @@
+  DependencyContainerManager._internal();
+
+  /// Initializes the app-level dependencies.
+  Future<void> init() async {
+    await _appLevelDependencies.init();
+    registerContainerFactory<AppLevelDependencyContainer>(
+        (_) => _appLevelDependencies);
+    subscribeToContainer<AppLevelDependencyContainer>(this);
+  }
+
+  /// Registers a dependency container factory for a specific type [T].
+  ///
+  /// This method allows you to register a factory function that will be responsible
+  /// for creating instances of the dependency container for the specified dependency container
+  /// type [T].
+  ///
+  /// [factory]: A function that takes an existing [DependencyContainer] and returns
+  /// a new instance of a [DependencyContainer] for the specified type [T].
+  void registerContainerFactory<T>(
+      DependencyContainer Function(DependencyContainer) factory) {
+    _containerFactories[T] = factory;
+  }
+
+  /// Adds a subscriber for the container of type [C] and manages the container's lifecycle.
+  ///
+  /// This method registers a subscriber for the specified container type [C] and
+  /// manages the lifecycle of the container based on the presence of subscribers.
+  ///
+  /// [subscriber]: An object that subscribes to the container of type [C]. The presence
+  /// of subscribers influences the creation and destruction of the container.
+  void subscribeToContainer<C>(Object subscriber) {
+    final currentSubscribers = _containerSubscribers[C] ??= {};
+    // Add the subscriber to the set of subscribers for the container type C if not already present.
+    if (currentSubscribers.add(subscriber)) {
+      _manageContainerLifecycle<C>();
+    }
+  }
+
+  /// Removes a subscriber for the container of type [C] and manages the container's lifecycle.
+  ///
+  /// This method unregisters a subscriber for the specified container type [C] and
+  /// manages the lifecycle of the container based on the remaining subscribers.
+  ///
+  /// [subscriber]: An object that unsubscribes from the container of type [C]. The absence
+  /// of subscribers can lead to the destruction of the container.
+  void unsubscribeFromContainer<C>(Object subscriber) {
+    final currentSubscribers = _containerSubscribers[C];
+    // Remove the subscriber from the set of subscribers for the container type C if present.
+    if (currentSubscribers != null && currentSubscribers.remove(subscriber)) {
+      _manageContainerLifecycle<C>();
+    }
+  }
+
+  /// Retrieves the container of type [C].
+  ///
+  /// This method returns the active container of the specified type [C]. If the container
+  /// does not exist, an error is thrown.
+  ///
+  /// Throws [StateError] if the container of type [C] does not exist.
+  ///
+  /// Returns an instance of the container of type [C].
+  C getContainer<C>() {
+    final container = _activeContainers[C];
+    if (container == null) {
+      throw StateError('Container of type $C does not exist.');
```

Comment:
Let's maybe provide some instruction with the error. Like: before a container can be resolved, there should be a subscriber... etc

Reply:
Gave hints here: Gave hints here: aadaa25bbcf498b82a7888be911e49b4120319a5